### DISCUSSION TITLE
Tab auto-complete does not seems to work.

### DISCUSSION THREAD

**Discussion by zxyao145** [kubernetes-client/csharp#1193] [Q&A] [answered]
When I press tab character, it takes tab as input white spaces instead of auto-completing, and if I use rancher, it works well

![image](https://user-images.githubusercontent.com/24973084/216668095-8fbcc5d8-80f8-4448-babd-af1d7703903b.png)

![image](https://user-images.githubusercontent.com/24973084/216668760-514dab49-0e5f-4e4f-9c72-7c4968d054c3.png)

And my code like this.
```c#
    var config = KubernetesClientConfiguration.BuildConfigFromConfigFile();
    IKubernetes client = new Kubernetes(config);

    var msgObj = new
    {
        PodName = "app-test-dev-55f447dfb5-jpznl",
        ContainerName = "ubuntu"
    };

    var webSocket =
        await client.WebSocketNamespacedPodExecAsync(
            msgObj.PodName, 
            "default",
            new List<string>()
            {
                "/bin/sh"
            },
            msgObj.ContainerName
            )
            .ConfigureAwait(false);

    var demux = new StreamDemuxer(webSocket);
    demux.Start();


    var streamIn = demux.GetStream(ChannelIndex.StdIn, ChannelIndex.StdIn);
    var streamOutput = demux.GetStream(ChannelIndex.StdOut, ChannelIndex.StdOut);

    var t1 = Task.Run(async () => await Console.OpenStandardInput().CopyToAsync(streamIn));
    var t2 = Task.Run(async () => await streamOutput.CopyToAsync(Console.OpenStandardOutput()));

    await Task.WhenAny(t1, t2);
```

Is there anything wrong with the code?

**Comment by tg123** [comment, marked as answer]
did you test `/bin/bash` ?
i do not think `sh` supports auto complete

**Reply by zxyao145** [reply]
> did you test `/bin/bash` ? i do not think `sh` supports auto complete

Sorry, `/bin/bash` doesn't work either.

**Reply by zxyao145** [reply]
Sorry for my bad test, it works well in aspnet core but not works in console app. Maybe `stream.CopyToAsync` make some some special handle.

Thans for your reply!

**Reply by tg123** [reply]
hmmm
i did not think it has something to do with asp.net. stream shared exactly the same runtime with console apps

**Reply by zxyao145** [reply]
Yes, but the current result is really like this. I don't know what happened. In the aspnet core, I use websocket to receive the request, parse the result, and then send then request to pod.
